### ISSUE TITLE
Amplify Publish fails with no useful information

### ISSUE THREAD

**Issue by rewardle-jason** [aws-amplify/amplify-console#1369]
**Describe the bug**

Attempting to run amplify publish on an already published amplify project.  The project it built successfully and the artifacts zipped completed all ok.  But the deploy fails with the following message:

✖ Deployment failed! Please report an issue on the Amplify Console GitHub issue tracker at https://github.com/aws-amplify/amplify-console/issues.
An error occurred during the publish operation

**Logs**
Below is the full console log leading up to the error:
jason@iOSs-MacBook-Pro HolterSoft % amplify publish      
✔ Successfully pulled backend environment sandbox from the cloud.

Current Environment: sandbox

| Category      | Resource name              | Operation | Provider plugin   |
| ------------- | -------------------------- | --------- | ----------------- |
| Auth          | userPoolGroups             | No Change | awscloudformation |
| Auth          | holtersoft7caa79297caa7929 | No Change | awscloudformation |
| Function      | AdminQueriesa57033f4       | No Change | awscloudformation |
| Function      | CreateClinicUser           | No Change | awscloudformation |
| Function      | filemanagement             | No Change | awscloudformation |
| Function      | S3Triggera08b3458          | No Change | awscloudformation |
| Function      | migrateholtercases         | No Change | awscloudformation |
| Function      | testitems                  | No Change | awscloudformation |
| Function      | migratecheckouts           | No Change | awscloudformation |
| Function      | migratecheckins            | No Change | awscloudformation |
| Function      | migratepatients            | No Change | awscloudformation |
| Api           | AdminQueries               | No Change | awscloudformation |
| Api           | holtersoft                 | No Change | awscloudformation |
| Api           | datamigration              | No Change | awscloudformation |
| Storage       | holtercasedocuments        | No Change | awscloudformation |
| Hosting       | amplifyhosting             | No Change | awscloudformation |
| Notifications | holtersofti9CSB            | No Change |                   |

No changes detected
Publish started for amplifyhosting

> holtersoft@0.1.0 build /Users/jasonpotter/Documents/Projects/HolterSoft
> react-scripts build

Creating an optimized production build...

File sizes after gzip:

  525.77 KB  build/static/js/3.7fdf7ceb.chunk.js
  64.45 KB   build/static/css/3.220ccb42.chunk.css
  26.38 KB   build/static/js/main.07c7886e.chunk.js
  24.98 KB   build/static/js/34.d48715b4.chunk.js
  13.33 KB   build/static/js/8.d965d6a6.chunk.js
  11.68 KB   build/static/js/4.28ff3531.chunk.js
  6.59 KB    build/static/js/14.db257351.chunk.js
  6.08 KB    build/static/js/7.0e37d79a.chunk.js
  6.08 KB    build/static/js/6.000c7d9d.chunk.js
  5.8 KB     build/static/js/15.351301f1.chunk.js
  3.62 KB    build/static/js/17.30511376.chunk.js
  3.34 KB    build/static/js/polyfills-css-shim.06abaf56.chunk.js
  3.2 KB     build/static/js/9.6ad60700.chunk.js
  2.91 KB    build/static/js/5.0ae38c03.chunk.js
  2.61 KB    build/static/js/16.77ab06d6.chunk.js
  2.15 KB    build/static/js/13.2424acc9.chunk.js
  2.11 KB    build/static/js/33.7e682862.chunk.js
  1.94 KB    build/static/js/11.c06e65e3.chunk.js
  1.9 KB     build/static/js/10.8b71c020.chunk.js
  1.83 KB    build/static/js/12.abb5abd8.chunk.js
  1.79 KB    build/static/js/30.26f1c46d.chunk.js
  1.57 KB    build/static/js/18.59ff0955.chunk.js
  1.48 KB    build/static/js/runtime-main.453bc936.js
  1.4 KB     build/static/js/35.d32a103a.chunk.js
  1.28 KB    build/static/js/27.aa9955ba.chunk.js
  1.11 KB    build/static/js/23.26ce96f3.chunk.js
  1.06 KB    build/static/js/28.dff44e93.chunk.js
  889 B      build/static/js/21.1b52a8e0.chunk.js
  843 B      build/static/js/31.f7633f5d.chunk.js
  702 B      build/static/js/32.d8ca8787.chunk.js
  652 B      build/static/js/25.ef85c121.chunk.js
  641 B      build/static/js/22.f8641923.chunk.js
  640 B      build/static/js/24.cbf17a38.chunk.js
  541 B      build/static/js/29.2df0d7df.chunk.js
  485 B      build/static/js/19.e34ec7d7.chunk.js
  451 B      build/static/js/26.a434cb52.chunk.js
  339 B      build/static/js/20.67ec0dfd.chunk.js
  322 B      build/static/css/main.c2ebcf18.chunk.css

The project was built assuming it is hosted at /.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.
You may serve it with a static server:

  yarn global add serve
  serve -s build

Find out more about deployment here:

  bit.ly/CRA-deploy

✔ Zipping artifacts completed.
✖ Deployment failed! Please report an issue on the Amplify Console GitHub issue tracker at https://github.com/aws-amplify/amplify-console/issues .
An error occurred during the publish operation
jason@iOSs-MacBook-Pro HolterSoft % 

**Expected behavior**
I would expect the deploy to work or give more information on the reason for the failure.

**Versions**
Amplify 4.39.0
OS  Mac

**Comment by rewardle-jason** [comment]
ok turned out to be a simple fix but more helpful error message would help.

I had not run the following command on my new machine. 
`amplify configure project`

 I just checked out the code from git and just tried to publish.

In this process I updated the aws profile settings.  Initially I had no config or credentials file in my ~/.aws folder.

A simple check for this before publish might help avoid this issue.
